New introductory blog post (series) (#10)
kolabri/kolabri.io: content/blog/2024-challenges/index.md

---

> title: 'Tax Compliance for Software Engineering: The Fundemental Challenges in Software Engineering'

Reviewer: ```suggestion
title: 'Tax Compliance for Software Engineering: The Fundemental Challenges'
```

> slug: 'what-is-it-all-about'

Reviewer: Die URL mit dem Slug zu modifizieren hat imho zwei Ziele: Die URL für Menschen gefälliger/lesbarer zu machen sowie SEO. Für SEO finde ich diesen Slug nicht optimal (für Menschen dafür prima!).

Um beides unter einen Hut zu bringen wäre mein Vorschlag entweder gar keinen manuellen Slug zu vergeben (lesbar, aber lang) oder alternativ einen SEO-freundlicheren Slug mit mehr Kontext zu wählen:

```suggestion
slug: 'tax-compliance-in-software-engineering-fundamental-challenges'
```

Reviewer: ⚠️ Bitte links nicht vergessen 🙂

Author: Wir können gerne einen längeren slug nehmen, aber challenges ist der zweite blog post. :) Ich würde also `tax-compliance-in-software-engineering-what-is-it-all-about` nehmen

> date: 2024-03-12T06:00:00+02:00

Reviewer: Um mehr Leben im Blog zu erzeugen (vorzutäuschen?) und die Reihenfolge expliziter zu kennzeichnen, würde ich diesen Blog-Post eine Woche nach dem anderen erscheinen lassen:

```suggestion
date: 2024-03-19T06:00:00+02:00
```

draft: false
description: "The second article of our series on tax compliance for software engineering reveals the fundemantal challenges unique to software engineering."
---
{{< post-author >}}
Michael Dorner
{{< /post-author >}}

> # Tax Compliance for Software Engineering: The Fundemental Challenges in Software Engineering

Reviewer: ```suggestion
# Tax Compliance for Software Engineering: The Fundemental Challenges
```

Modern software systems are often too large, too complex, and evolving too fast for single developers to oversee. Therefore, software engineering has become highly collaborative. Often, engineers and their teams collaborate across borders, especially in multinational companies with subsidiaries spread around the globe. This collaboration has an often overlooked legal implication if individuals collaborate across borders: the profits from those cross-border collaborations become taxable.

> In this second blog post in our [series on tax compliance for software engineering](/content/blog/2024-what-is-it-all-about/_index.md), we will discuss the unique and fundamental challenges of taxing collaborative software engineering. In our the last article of our series we will illustrate the industrial significance of cross-border collaboration in an industrial case, namely code review.

Reviewer: 👍 Sehr clever mit dem Eingangs-Paragraphen der immer gleich bleibt und anschließend der Überleitung, zur Blog-Post-Serie.

> So, what are transfer prices for collaborative software engineering that comply with this arm's length principle? Determining a market price for intangibles is inherently difficult and is reflected in a broad price range. Collaborative software engineering, however, scales the problem of a transfer-price determination to a new level of complexity because the reality of modern software engineering is significantly more complex than our introductory example above may suggest. Since transfer price regulations apply to a much broader definition of intangibles compared to accounting standards, the latter can not be used as a reliable measure of value for transfer pricing purposes.

Reviewer: Klar, wir haben im letzten Blog-Post das `Arm's Length Principle` erklärt. Wir wissen aber nicht, wie die Leute auf diesen (den zweiten) Blog-Post aufmerksam geworden sind. Deshalb würde ich wohl noch einen Erklärsatz hinzufügen:

```suggestion
Associated enterprises are required to behave and charge each other as if they were independent third parties - the so called Arm's Length Principle. So, what are transfer prices for typical transactions in collaborative software engineering that comply with this principle? Determining a market price for intangibles is inherently difficult and is reflected in a broad price range. Collaborative software engineering, however, scales the problem of a transfer-price determination to a new level of complexity because the reality of modern software engineering is significantly more complex than our introductory example above may suggest. Since transfer price regulations apply to a much broader definition of intangibles compared to accounting standards, the latter can not be used as a reliable measure of value for transfer pricing purposes.
```

> From our last [blog post](/content/blog/2024-what-is-it-all-about/_index.md), we understood that companies must pay a transfer price for goods, intangibles, or services transferred in a multinational enterprise to comply with the international standards on taxation. So, what are transfer prices for collaborative software engineering that comply with this arm's length principle? Determining a market price for intangibles is inherently difficult and is reflected in a broad price range. Collaborative software engineering, however, scales the problem of a transfer-price determination to a new level of complexity because the reality of modern software engineering is significantly more complex than our introductory example above may suggest. Since transfer price regulations apply to a much broader definition of intangibles compared to accounting standards, the latter can not be used as a reliable measure of value for transfer pricing purposes.

Reviewer: Der Paragraph scheint mir irgendwie eine Dopplung zum vorherigen (der allerdings schon einen Verweis auf den letzten Blog-Post enthält). Kann einer der beiden Paragraphen entfallen?

> In the following, we discuss three main high-level challenges for transfer pricing in collaborative software engineering within multinational enterprises. The figure below highlights the complexity in modern collaborative software engineering at *devnullsoft Group* (the fictitious multinational company from [our last post](/content/blog/2024-what-is-it-all-about/_index.md)) and where those three challenges apply.

Reviewer: ```suggestion
There are three high-level challenges for transfer pricing in collaborative software engineering within multinational enterprises. The figure below illustrates the complexity in modern collaborative software engineering at *devnullsoft Group* (the fictitious multinational company introduced in [the previous post of this series](/content/blog/2024-what-is-it-all-about/_index.md)) and where those three challenges apply.
```

![The challenges for tax compliance in modern collaborative software engineering](challenges.png)

## Challenge 1: What is a taxable transaction in software engineering?

> The trouble for transfer pricing in software engineering begins with a fundamental question: What is actually a taxable transaction in the context of collaborative software engineering? We simply do not know what types or characteristics types or characteristics classify a taxable exchange of intangibles or services across the boundaries of a country in the context of software engineering.

Reviewer: ```suggestion
The trouble for transfer pricing in software engineering begins with a fundamental question: What is actually a taxable transaction in the context of collaborative software engineering? It is not immediately clear what types or characteristics classify a taxable exchange of intangibles or services across the boundaries of a country in the context of software engineering.
```

> Among other potentially relevant types of taxable intangibles, such as *goodwill* or *group synergies*, we discuss in this and the following subsections two types of intangibles that are highly relevant for software engineering: *know-how* and *licenses*.

Reviewer: ```suggestion
Among other potentially relevant types of taxable intangibles, such as *goodwill* or *group synergies*, this blog post focusses on two types of intangibles that are highly relevant for software engineering: *know-how* and *licenses*.
```

> Additionally, we see an interplay between those two types of intangibles, know-how and licensing, since they may be two sides of the same collaboration. For example, when code contributions from the component user support instance of reuse.

Reviewer: ```suggestion
Additionally, we see an interplay between those two types of intangibles, know-how and licensing, since they may be two sides of the same collaboration. For example, when code contributions from a component user support instances of reuse.
```

2. Privacy: Analyzing the complete communication of developers may be perceived as a measure of surveillance, which raises ethical and legal concerns related to privacy.

In contrast to the potentially rich sources for tracing taxable transactions from collaboration tools used in software engineering, tracking company-internal reuse often lacks a solid data basis. Although companies often track the reuse of external open-source components for open-source license compliance purposes, those tools are rarely used or suitable for tracking company-internal reuse. Also, only reuse that crosses borders is taxable; information that is often not available or stored over time, although component ownership is not static and may be subject to change.

**Insight: Data for tracking taxable transactions may be incomplete, faulty with respect to location, or restricted. There is no dedicated tool support yet for the practical transfer price determination.**

## Challenge 3: How to value taxable transactions in software engineering?

> While it is inherently difficult to tax intangibles in general, things are even more complicated in software engineering. Potentially taxable intangibles cover a large range of granularity in software engineering: They may be as large as a microservice providing user authentication used by microservices of other teams (see intangible *licenses*) or as small as a code change, code review, or bug report (see intangible *know-how*). Although the code change or feedback in a code review is small-maybe even only one line of code, like in the case of the *Heartbleed* security bug in the OpenSSL cryptography library from 2014-the potential impact on the software system can be tremendous or even fatal. A software change or a code review delivers value through impact, not size.

Reviewer: ```suggestion
While it is inherently difficult to tax intangibles in general, things are even more complicated in software engineering. Potentially taxable intangibles cover a large range of granularity: They may be as large as a microservice providing user authentication which itself is used by microservices of other teams (see intangible *licenses*) or as small as a code change, code review, or bug report (see intangible *know-how*). Although the code change or feedback in a code review is small - maybe even only one line of code, like in the case of the *Heartbleed* security bug in the OpenSSL cryptography library from 2014-the potential impact on the software system can be tremendous or even fatal. A software change or a code review delivers value through impact, not size.
```

The same applies to licensing. The number of use relations of a software component or its size (however defined) does not reflect the value provided to the software-intensive product. While the software component for user authentication may be important for operating the software-intensive product and, therefore, has a large amount of depedent software components, it is not differentiating and may even be considered a commodity.

This means we cannot simply use purely quantitative measurements for transfer pricing. However, the sheer mass of small, fine-grained transactions of all types makes a human qualitative case-by-case evaluation impossible.

**Insight: A purely quantitative valuing can hardly reflect the value of transactions; however, a purely qualitative assessment does not scale with the magnitude of cross-border transactions in modern software development. Transfer pricing requires a tool-supported processes that assists tax and software engineering experts.**

## Conclusion

> If you want to learn more about how we at Kolabri addresses those challenges, read our brochure on how we leverage data analytics for tax compliance in software engineering. If you want to learn more about the prevalence of cross-border collaboration in modern collaborative software engineering and how we measured it, read [the next blog post in our series on tax compliance for software engineering]().

Reviewer: ```suggestion
If you want to learn more about how we at Kolabri address those challenges, read our brochure on data analytics for tax compliance in software engineering. If you want to learn more about the prevalence of cross-border collaboration in modern collaborative software engineering and how we measured it, stay tuned for the next blog post in our series on tax compliance for software engineering.
```
